### ISSUE TITLE
New Rule: `no-empty-definitions`

### ISSUE THREAD

**Issue by lumirlumir** [eslint/markdown#359]
### Rule details

Rule to disallow empty definitions

### What type of rule is this?

Warns about a potential problem

### Example code

```markdown
<!-- OK -->

[earth]: https://example.com/earth/ // Valid

<!-- ERROR -->

[earth]: <> // Error
[earth]: // Error
```

### Participation

- [x] I am willing to submit a pull request to implement this rule.

### Additional comments

Hello,

I'd like to propose a new rule called `no-empty-definitions`.

This suggestion arose while I was working on a PR for the `no-empty-images` rule.
(Reference: [https://github.com/eslint/markdown/pull/357#discussion\_r2073168298](https://github.com/eslint/markdown/pull/357#discussion_r2073168298))

Currently, the [`remark-lint-no-empty-url`](https://github.com/remarkjs/remark-lint/tree/main/packages/remark-lint-no-empty-url#examples) rule handles a mix of different concerns — specifically `no-empty-links`, `no-empty-images`, and `no-empty-definitions`.

* `no-empty-links` is already implemented in `@eslint/markdown`
* `no-empty-images` is currently under development
* `no-empty-definitions` is what I’m proposing here

Initially, I considered unifying these into a single rule, such as `no-empty-urls`. However, doing so would likely require merging `no-empty-links` into it as well, since their logic would overlap. This, in turn, could introduce a breaking change.

To avoid that, I’d like to suggest creating a separate rule specifically for empty definitions: `no-empty-definitions`. This approach keeps the responsibilities clearly separated and avoids introducing any breaking changes to existing rules.

---

If this suggestion is accepted, I’d be happy if you could assign it to me! I’ll make sure to complete it by May 14.

**Comment by nzakas** [comment]
I think this makes sense. :+1:

**Comment by nzakas** [comment]
That's up to @lumirlumir.

**Comment by lumirlumir** [comment]
@Pixel998 

If you'd prefer to take it, feel free — I'm happy to step aside.

But please note that [earth]: would be recognized as a normal text node, not a definition node. So, we need to handle this situation.
